ledger-core/v2/virtual/object/object.go

```diff
@@ -321,6 +338,32 @@ func (sm *SMObject) stepWaitIndefinitely(ctx smachine.ExecutionContext) smachine
 	return ctx.Sleep().ThenRepeat()
 }
 
+func (sm *SMObject) createWaitOrderedPendingTableSM(ctx smachine.ExecutionContext) {
+	syncSM := SMAwaitTableFill{
+		sync: sm.OrderedPendingListFilled,
+	}
+
+	// syncSM acquire OrderedExecute semaphore in init step.
+	ctx.InitChildWithPostInit(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+		return &syncSM
+	}, func() {
+		sm.OrderedPendingListFilledCallback = syncSM.stop
+	})
+}
+
+func (sm *SMObject) createWaitUnorderedPendingTableSM(ctx smachine.ExecutionContext) {
+	syncSM := SMAwaitTableFill{
+		sync: sm.UnorderedPendingListFilled,
+	}
+
+	// syncSM acquire OrderedExecute semaphore in init step.
+	ctx.InitChildWithPostInit(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+		return &syncSM
+	}, func() {
+		sm.UnorderedPendingListFilledCallback = syncSM.stop
+	})
```

Comment:
I'd suggest checking of result of `InitChild`. I'm not completely sure what'll happened if child SM will panic.

Reply:
But passing nil to bargein will require some backtracking, when we'll debug that code.

Reply:
`InitChild` will either create child or panic, as far as I see the code

Reply:
I can't see part with panicking, otherwise I see https://github.com/insolar/assured-ledger/blob/ce7934b73beea3480fa0c57eb39600dccddfe17d/ledger-core/v2/conveyor/smachine/slot_machine_execute.go#L299-L303 that part.

Reply:
I mean InitChildWithPostInit - https://github.com/insolar/assured-ledger/blob/c4b5abcd4b9699d5dd934dd3c54626a2791b1c52/ledger-core/v2/conveyor/smachine/context_basic.go#L228 it either adds SM, or panics. postInitFn should either work, or panic, there are no results

Reply:
Nope, I've added panic into `SMAwaitDelegate` and got that:
```
=== RUN   TestSMObject_Semi_CheckAwaitDelegateIsStarted
2020-06-01T15:41:24.269402000+03:00 LOG 0xc000314000[  1] 001 @ 001: jump current=<init> next=(*SMObject).stepGetState timing=1ns/93µs payload=*object.SMObject tracer=
2020-06-01T15:41:24.269587000+03:00 LOG 0xc000314000[  2] 001 @ 003: jumpLoop (short) current=(*SMObject).stepGetState next=(*SMObject).stepGotState timing=3µs/4µs payload=*object.SMObject tracer=
2020-06-01T15:41:24.269855000+03:00 LOG 0xc000314000[  2] 002 @ 001: panic (inline) current=<init> next=<nil> payload=*object.SMAwaitDelegate tracer=
2020-06-01T15:41:24.269873000+03:00 ERR 0xc000314000[  2] 002 @ 001: panic current=<init> next=<nil> timing=1ns/1ns payload=*object.SMAwaitDelegate tracer= errorMsg=initialization: 123
Stack trace:
panic(0x17cb0e0, 0x1a2dd70)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/insolar/assured-ledger/ledger-core/v2/virtual/object.(*SMAwaitDelegate).Init(0xc0000ef0a0, 0x1a5db20, 0xc00000e3c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/data/go/src/github.com/insolar/assured-ledger/ledger-core/v2/virtual/object/awaitdelegate.go:27 +0x58
github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine.(*initializationContext).executeInitialization(0xc00000e3c0, 0xc0000ea6e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/data/go/src/github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine/context_misc.go:122 +0xd4
<panic>
    TestSMObject_Semi_CheckAwaitDelegateIsStarted: redirector.go:75: {<panic>}
2020-06-01T15:41:24.270365000+03:00 ERR 0xc000314000[  2] 002 @ 001: internal recycle current=<init> payload=*object.SMAwaitDelegate tracer= errorMsg=initialization: 123
Stack trace:
panic(0x17cb0e0, 0x1a2dd70)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/insolar/assured-ledger/ledger-core/v2/virtual/object.(*SMAwaitDelegate).Init(0xc0000ef0a0, 0x1a5db20, 0xc00000e3c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/data/go/src/github.com/insolar/assured-ledger/ledger-core/v2/virtual/object/awaitdelegate.go:27 +0x58
github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine.(*initializationContext).executeInitialization(0xc00000e3c0, 0xc0000ea6e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/data/go/src/github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine/context_misc.go:122 +0xd4
<panic>
2020-06-01T15:41:24.270727000+03:00 LOG 0xc000314000[  2] 001 @ 004: jumpLoop (short) current=(*SMObject).stepGotState next=(*SMObject).stepReadyToWork timing=1ns/1.14ms payload=*object.SMObject tracer=
2020-06-01T15:41:24.270745000+03:00 LOG 0xc000314000[  2] 001 @ 004: repeat current=(*SMObject).stepReadyToWork next=<nil> timing=1ns/19µs payload=*object.SMObject tracer=
--- PASS: TestSMObject_Semi_CheckAwaitDelegateIsStarted (0.00s)
PASS
```

Reply:
ok, added IsZero check for bargeIN

Reply:
reverted check for empty bargeIn, will create separate ticket for this. Right now this behaviour breaks unit tests and could not be fixed without major changes around BargeIn object